src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/CngKey.Create.cs

```diff
@@ -149,6 +159,114 @@ private static void InitializeKeyUiPolicyProperties(SafeNCryptKeyHandle keyHandl
+                //     byte[cbFieldLength]          G.Y
+                //     byte[cbSubgroupOrder]        Order (n)
+                //     byte[cbCofactor]             Cofactor (h)
+                //     byte[cbSeed]                 Seed
+
+                int blobSize = sizeof(BCRYPT_ECC_PARAMETER_HEADER) +
+                    curve.Prime.Length +
+                    curve.A.Length +
+                    curve.B.Length +
+                    curve.G.X.Length +
+                    curve.G.Y.Length +
+                    curve.Order.Length +
+                    curve.Cofactor.Length +
+                    (curve.Seed == null ? 0 : curve.Seed.Length);
+
+                byte[] blob = new byte[blobSize];
+                fixed (byte* pBlob = blob)
+                {
+                    // Build the header
+                    BCRYPT_ECC_PARAMETER_HEADER* pBcryptBlob = (BCRYPT_ECC_PARAMETER_HEADER*)pBlob;
+                    pBcryptBlob->Version = Interop.BCrypt.BCRYPT_ECC_PARAMETER_HEADER_V1;
+                    pBcryptBlob->cbCofactor = curve.Cofactor.Length;
+                    pBcryptBlob->cbFieldLength = curve.A.Length; // P, A, B, X, Y have the same length
+                    pBcryptBlob->cbSeed = curve.Seed == null ? 0 : curve.Seed.Length;
+                    pBcryptBlob->cbSubgroupOrder = curve.Order.Length;
+                    pBcryptBlob->CurveGenerationAlgId = ECCng.GetHashAlgorithmId(curve.Hash);
+                    pBcryptBlob->CurveType = (Interop.BCrypt.ECC_CURVE_TYPE_ENUM)curve.CurveType;
+
+                    // Emit the blob
+                    int offset = sizeof(BCRYPT_ECC_PARAMETER_HEADER);
+                    Interop.BCrypt.Emit(blob, ref offset, curve.Prime);
+                    Interop.BCrypt.Emit(blob, ref offset, curve.A);
+                    Interop.BCrypt.Emit(blob, ref offset, curve.B);
+                    Interop.BCrypt.Emit(blob, ref offset, curve.G.X);
+                    Interop.BCrypt.Emit(blob, ref offset, curve.G.Y);
+                    Interop.BCrypt.Emit(blob, ref offset, curve.Order);
+                    Interop.BCrypt.Emit(blob, ref offset, curve.Cofactor);
+                    if (curve.Seed != null)
+                    {
+                        Interop.BCrypt.Emit(blob, ref offset, curve.Seed);
+                    }
+
+                    // We better have computed the right allocation size above!
+                    Debug.Assert(offset == blobSize, "offset == blobSize");
+
+                    return new CngProperty(
+                        Interop.BCrypt.BCryptPropertyStrings.BCRYPT_ECC_PARAMETERS, blob, 
+                        CngPropertyOptions.None);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Map a curve name to algorithm. This enables curves that worked pre-Win10
+        /// to work with newer APIs for import and export.
+        /// </summary>
+        internal static CngAlgorithm CurveNameToAlgorithm(string name)
```

Comment:
This should probably have ECDsa somewhere in the name, since we'll be needing a similar thing for ECDH when we add that.